pkg/search/postgres/store.go

```diff
@@ -171,10 +172,16 @@ func (s *genericStore[T, PT]) Exists(ctx context.Context, id string) (bool, erro
 }
 
 // Count returns the number of objects in the store.
-func (s *genericStore[T, PT]) Count(ctx context.Context) (int, error) {
+func (s *genericStore[T, PT]) Count(ctx context.Context, q *v1.Query) (int, error) {
 	defer s.setPostgresOperationDurationTime(time.Now(), ops.Count)
 
-	return RunCountRequestForSchema(ctx, s.schema, search.EmptyQuery(), s.db)
+	return RunCountRequestForSchema(ctx, s.schema, q, s.db)
+}
+
+func (s *genericStore[T, PT]) Search(ctx context.Context, q *v1.Query) ([]search.Result, error) {
+	defer s.setPostgresOperationDurationTime(time.Now(), ops.Search)
+
+	return RunSearchRequestForSchema(ctx, s.schema, q, s.db)
```

Comment:
AFAIK the indexer generation depends on the options map that is embedded within the schema. Therefore, it is guaranteed because this PR does not change the option maps associated with the schema nor the schema generation but only the store generation. 

I ask you to elaborate on your comment if I did not address your concern, otherwise resolve the thread.